### DISCUSSION TITLE
Redirect Loop when using subdomains

### DISCUSSION THREAD

**Discussion by ghost** [coder/coder#15708] [General] [answered]
I have coder installed locally in talos kubernetes and have a port forward done in kubernetes so i can access it for testing in the browser. This works and i can login. But when I want to access a app that has subdomain = true enabled (in this cas kasm vnc), then i get a redirect loop.

![redirect_1](https://github.com/user-attachments/assets/666c11d2-2726-4e29-93f5-01048dd0d48d)
![redirect_2](https://github.com/user-attachments/assets/03af8236-024a-44ea-b7c9-7cfa4d6e6c0b)
![redirect_3](https://github.com/user-attachments/assets/195cf6e4-146d-4e20-bbcb-2747197a07d7)
![redirect_4](https://github.com/user-attachments/assets/9a0518f9-7e88-443f-b66b-ef72ce0ac028)
![redirect_5](https://github.com/user-attachments/assets/76be691a-9ffc-41cd-abd2-e3011ba42c9d)
![image](https://github.com/user-attachments/assets/7afbed58-0461-4c0a-a543-53af199fe99e)
![image](https://github.com/user-attachments/assets/f4610bc5-974b-4c0f-b2b4-77375bb47cd6)

It seems like http://kasm-vnc--main--test--test.kasm/ is not sending any cookie even after it was redirected to authentication. How can the Domain be set to a wildcard?

**Comment by spikecurtis** [comment]
Have you set `CODER_WILDCARD_ACCESS_URL`?  [Coder docs](https://coder.com/docs/admin/setup#wildcard-access-url)

**Reply by ghost** [reply]
yes.
  {
                                name: "CODER_ACCESS_URL",
                                value: "http://kasm"
                            },
                            {
                                name: "CODER_WILDCARD_ACCESS_URL",
                                value: "*.kasm"
                            },

**Reply by spikecurtis** [reply, marked as answer]
It looks like your browser is blocking setting the cookie.  The Origin setting the cookie is `kasm-vnc--main--test--test.kasm`, and the cookie domain is `kasm`, which I would expect to be allowed, since `kasm` is the parent domain. However, this isn't allowed if the parent domain is a "public" domain like `.com` or `.co.uk`.  I wonder whether `kasm` is being rejected because it's being used as a TLD in this case? You might try with something like `coder.kasm` and `*.coder.kasm` instead.

**Reply by ghost** [reply]
yes that was it thanks a lot.

**Reply by ghost** [reply]
Should be added to coder docu
